src/masonite/tests/HttpTestResponse.py

```diff
@@ -37,8 +52,221 @@ def assertIsNotNamed(self, name=None):
+        """Assert that view context contains a given data key (and eventually associated value)."""
+        self._ensure_response_has_view()
+        assert key in self.response.original.dictionary
+        if value:
+            assert self.response.original.dictionary[key] == value
+        return self
+
+    def assertViewHasAll(self, keys):
+        """Assert that view context contains exactly the data keys (or the complete data dict)."""
+        self._ensure_response_has_view()
+        if isinstance(keys, list):
+            assert set(keys) == set(self.response.original.dictionary.keys()) - set(self.response.original._shared.keys())
+        else:
+            view_data = self.response.original.dictionary
+            for key in self.response.original._shared:
+                del view_data[key]
+            assert keys == view_data
+        return self
+
+    def assertViewMissing(self, key):
+        """Assert that given data key is not in the view context."""
+        self._ensure_response_has_view()
+        assert key not in self.response.original.dictionary
+        return self
+
+    def assertAuthenticated(self):
+        assert self.application.make("auth").guard("web").user()
+        return self
+
+    def assertGuest(self):
+        assert not self.application.make("auth").guard("web").user()
+        return self
+
+    def assertAuthenticatedAs(self, user):
+        user = self.application.make("auth").guard("web").user()
+        assert user == user
+        return self
+
+    def assertHasHttpMiddleware(self, middleware):
+        """Assert that the request/response cycle has the given middleware. The HTTP middleware
+        class should be given."""
+        assert middleware in self.application.make("middleware").http_middleware
+        return self
+
+    def assertHasRouteMiddleware(self, middleware):
+        """Assert that the route has the given middleware. The registration key of the middleware
+        should be used."""
+        assert middleware in self.application.make("middleware").route_middleware
+        return self
+
+    def assertHasController(self, controller):
+        """Assert that route used the given controller. The controller can be a class or
+        a string. If it's a string it should be formatted as follow: ControllerName@method"""
+        if isinstance(controller, str) and "@" in controller:
+            assert self.route.controller == controller
+        elif issubclass(controller, Controller):
+            assert self.route.controller_class == controller
+        else:
+            raise ValueError("controller must be a string like YourController@index or a Controller class")
+        return self
+
+    def assertRouteHasParameter(self, key, value=None):
+        assert key in self.route.url_list, "Route does not contain parameter {key}."
+        if value is not None:
+            # TODO
+            # @josephmancuso not sure how to check if the route parameter has the given value
```

Comment:
here maybe for now I can remove the value check as I don"t see a solution and maybe it's not worth it. We could use `assertLocation` here to check the full path

Reply:
Hmm yeah not really possible right now I don't think. Maybe we can set it on the route class when we are compiling the route ..?